src/lint/track_config.nim

```diff
@@ -66,6 +66,46 @@ proc hasValidOnlineEditor(data: JsonNode, path: string): bool =
     ]
     result = allTrue(checks)
 
+const
+  statuses = ["wip", "beta", "active", "deprecated"].toHashSet()
+
+proc isValidConceptExercise(data: JsonNode; context, path: string): bool =
+  if isObject(data, context, path):
+    let checks = [
+      hasString(data, "slug", path),
+      hasString(data, "name", path),
+      hasString(data, "uuid", path),
+      hasBoolean(data, "deprecated", path, isRequired = false),
+      hasArrayOfStrings(data, "", "concepts", path),
+      hasArrayOfStrings(data, "", "prerequisites", path),
+      hasString(data, "status", path, isRequired = false, allowed = statuses),
+    ]
+    result = allTrue(checks)
+
+proc isValidPracticeExercise(data: JsonNode; context, path: string): bool =
+  if isObject(data, context, path):
+    let checks = [
+      hasString(data, "slug", path),
+      hasString(data, "name", path),
+      hasString(data, "uuid", path),
+      hasBoolean(data, "deprecated", path, isRequired = false),
+      hasInteger(data, "difficulty", path, allowed = 0..10),
+      if false: hasArrayOfStrings(data, "", "practices", path) else: true,
+      if false: hasArrayOfStrings(data, "", "prerequisites", path) else: true,
```

Comment:
Why are these checks disabled? Do they break too many tracks?

Reply:
Yes, with the current implementation of error messages in `configlet lint`, the output of `configlet lint` for nearly every track would be spammed with
```diff
+Array is empty: 'practices':
+./config.json
+
+Array is empty: 'prerequisites':
+./config.json
```
multiplied by the number of practice exercises, which is a lot of output. (Of course the error message should have more context like `exercise.practice[].practices` instead) .

The only tracks that do well on this check are:
```
- common-lisp
- elixir
- haxe
- zig
```

And to a lesser extent
```
- csharp
- python
```

It looks like the other tracks haven't populated them at all.

As of right now, running in a directory that contains every track repo:
```shell

$ rg '"practices": \[\]' | wc -l
3123
$ rg '"prerequisites": \[\]' | wc -l
3252
$ rg --multiline '"practices": \[\n' | wc -l
156
$ rg --multiline '"prerequisites": \[\n' | wc -l
393
```

That is, Exercism-wide, something like 90-95% of these arrays are empty.

| track           | count `"practices": []` | count `"prerequisites": []` | 
| --------------- | ----------------------- | --------------------------- | 
| arm64-assembly  |                       1 |                           1 | 
| babashka        |                      76 |                          76 | 
| ballerina       |                       9 |                           9 | 
| bash            |                      88 |                          89 | 
| c               |                      71 |                          71 | 
| ceylon          |                       8 |                           8 | 
| cfml            |                      28 |                          28 | 
| clojure         |                      76 |                          77 | 
| clojurescript   |                      21 |                          21 | 
| coffeescript    |                      21 |                          21 | 
| **common-lisp** |                       0 |                           1 | 
| coq             |                       3 |                           3 | 
| cpp             |                      55 |                          55 | 
| crystal         |                      50 |                          50 | 
| csharp          |                     117 |                         118 | 
| d               |                      34 |                          34 | 
| dart            |                      37 |                          38 | 
| delphi          |                      76 |                          76 | 
| **elixir**      |                       1 |                           2 | 
| elm             |                      46 |                          47 | 
| emacs-lisp      |                      28 |                          28 | 
| erlang          |                      77 |                          77 | 
| factor          |                       4 |                           4 | 
| fortran         |                      11 |                          11 | 
| fsharp          |                     123 |                         124 | 
| gleam           |                       5 |                           5 | 
| gnu-apl         |                       7 |                           7 | 
| go              |                     114 |                         115 | 
| groovy          |                      55 |                          55 | 
| **haxe**        |                       0 |                           0 | 
| haskell         |                     101 |                         101 | 
| idris           |                       5 |                           5 | 
| j               |                       7 |                           7 | 
| java            |                     120 |                         121 | 
| javascript      |                     112 |                         113 | 
| julia           |                      53 |                          54 | 
| kotlin          |                      88 |                          89 | 
| lfe             |                      32 |                          32 | 
| lua             |                      83 |                          83 | 
| mips            |                      14 |                          14 | 
| nim             |                      68 |                          68 | 
| objective-c     |                      51 |                          51 | 
| ocaml           |                      45 |                          45 | 
| perl5           |                      66 |                          66 | 
| pharo-smalltalk |                      42 |                          42 | 
| php             |                      66 |                          66 | 
| plsql           |                      11 |                          11 | 
| pony            |                      11 |                          11 | 
| powershell      |                      10 |                          10 | 
| prolog          |                      19 |                          19 | 
| purescript      |                      28 |                          29 | 
| **python**      |                      13 |                          14 | 
| r               |                      35 |                          35 | 
| racket          |                      34 |                          34 | 
| raku            |                      32 |                          32 | 
| reasonml        |                      24 |                          24 | 
| ruby            |                     103 |                         104 | 
| rust            |                      96 |                          97 | 
| scala           |                      95 |                          95 | 
| scheme          |                      41 |                          41 | 
| shen            |                       1 |                           1 | 
| sml             |                      28 |                          28 | 
| solidity        |                       3 |                           3 | 
| swift           |                      87 |                          88 | 
| tcl             |                     120 |                         120 | 
| typescript      |                      91 |                          91 | 
| vbnet           |                      10 |                          10 | 
| vimscript       |                      20 |                          20 | 
| x86-64-assembly |                      16 |                          17 | 
| **zig**         |                       0 |                           0 |

Reply:
Right. Let's keep them disabled for now then

Reply:
I've rebased this PR on another commit that allows us to suppress the errors for an empty array without `if false` - see [the force-push diff](https://github.com/exercism/configlet/compare/f81e476adcd52b68bfed057d8b831004369e8826..7bbd2a3e011aaa825cfebff715b7a169b6c188ec#diff-102d5f2912d17ee5540fbbbc4884ab571929bffc35a5d9c4b24dbe9bca240d32).